venv/pysnes/cpu.py

```diff
@@ -768,42 +779,44 @@ def fetch_decode_execute(self, code):
             self.PC = self.PC + 1
         # LDX dp
         elif opcode == 0xA6:
-            addr = self.fetch_byte(code)
-            wrapped_addr = self.compute_wrapped_addr(addr + self.DP)  # direct page wrapping
-            value = self.read_memory(wrapped_addr, byte_num = 2 - self.m(), wrapp=True) # zero bank wrapping!
+            byte = self.fetch_byte(code)
+            address = compute_addr.dp(byte, self.DP)
+            value = self.read_memory(address, byte_num = 2 - self.m(), wrapp=True) # zero bank wrapping!
             self.compute_flags(value, self.isX())
             self.X = value
             self.cycles += 4 - self.x() + self.w()
             self.PC = self.PC + 1
         # LDX abs
         elif opcode == 0xAE:
-            addr = self.fetch_twobyte(code) # no wrapping
-            value = self.read_memory((self.DBR << 16) + addr, byte_num = 2 - self.m())
+            bytes = self.fetch_twobyte(code)
+            address = compute_addr.abs(bytes, self.DBR)
+            value = self.read_memory(address, byte_num = 2 - self.m())
             self.compute_flags(value, self.isX())
             self.X = value
             self.cycles += 5 - self.x()
             self.PC = self.PC + 1
         # LDX dp, Y
         elif opcode == 0xB6:
-            addr = self.fetch_byte(code)
-            wrapped_addr = self.compute_wrapped_addr(addr + self.DP + self.Y)
-            value = self.read_memory(wrapped_addr,  byte_num = 2 - self.m(), wrapp=True) # zero bank wrapping!
+            byte = self.fetch_byte(code)
+            address = compute_addr.dp_x(byte, self.DP, self.Y, self.isX())
```

Comment:
I think compute_addr.dp_y should be used here

Reply:
Well never mind this was fixed in the "ISSUE44 Refactor AND, EOR, ORA using new addr computation" commit